app/src/main/java/com/droidhats/campuscompass/repositories/IndoorLocationRepository.kt

```diff
@@ -40,36 +43,40 @@ class IndoorLocationRepository private constructor(private val indoorLocationDao
             indoorLocationDao.deleteAllIndoor()
         }
         if (config == "debug" || indoorLocationDao.getOne().value == null) {
-            map.forEachBuilding { building ->
-                insertClasses(context, building)
+            for ((index, building) in map.getBuildings().withIndex()) {
+                insertClasses(context, building, index)
             }
         }
     }
 
-    fun insertClasses(context: Context, building: Building) {
+    private fun insertClasses(context: Context, building: Building, index: Int) {
 
         for (floorMap in building.getIndoorInfo().second) {
-            val inputStream: InputStream = context.assets.open(floorMap)
+            val inputStream: InputStream = context.assets.open(floorMap.value)
             val file: String = inputStream.bufferedReader().use { it.readText() }
             val mapProcessor: ProcessMap = ProcessMap()
             mapProcessor.readSVGFromString(file)
             val classes = mapProcessor.getClasses()
 
-            // todo: Consider the case where floor number is more than 1 digit
```

Comment:
We need to fix this todo before the merge

Reply:
Fixed

Reply:
It was actually already implemented, I just had forgotten to remove the todo